box-content-sdk/src/test/java/com/box/androidsdk/content/requests/BoxDownloadRequestTest.java

```diff
@@ -0,0 +1,119 @@
+package com.box.androidsdk.content.requests;
+
+import android.content.Context;
+
+import com.box.androidsdk.content.BoxApiFile;
+import com.box.androidsdk.content.listeners.DownloadStartListener;
+import com.box.androidsdk.content.listeners.ProgressListener;
+import com.box.androidsdk.content.models.BoxDownload;
+import com.box.androidsdk.content.testUtil.PowerMock;
+import com.box.androidsdk.content.testUtil.SessionUtil;
+
+import junit.framework.Assert;
+
+import org.junit.Test;
+import org.mockito.Mock;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Date;
+
+/**
+ * Tests for download requests
+ */
+
+@PrepareForTest({ BoxHttpResponse.class, BoxRequestDownload.class, BoxHttpRequest.class})
+public class BoxDownloadRequestTest extends PowerMock {
+    @Mock
+    Context mMockContext;
+    private long mBytesRead;
+    final static String DOWNLOAD_CONTENT = "testFileContent";
+    final static String FILE_ID = "5000948880";
+
+
+
+    @Test
+    public void testDownloadFileRequest() throws Exception {
+        BoxApiFile fileApi = new BoxApiFile(SessionUtil.newMockBoxSession(mMockContext));
+
+        //Byte stream to capture the outputstream to the server
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream(1024);
+        final Date now = new Date(System.currentTimeMillis());
+        final int contentLength = DOWNLOAD_CONTENT.getBytes().length;
+
+        //Mock httpurlconnection for downloads
+        URL u = PowerMockito.mock(URL.class);
+        PowerMockito.whenNew(URL.class).withAnyArguments().thenReturn(u);
+        HttpURLConnection huc = PowerMockito.mock(HttpURLConnection.class);
+        PowerMockito.when(u.openConnection()).thenReturn(huc);
+        PowerMockito.when(huc.getOutputStream()).thenReturn(outputStream);
+        PowerMockito.when(huc.getContentType()).thenReturn(BoxRequest.ContentTypes.JSON.toString());
+        PowerMockito.when(huc.getResponseCode()).thenReturn(200);
+        PowerMockito.when(huc.getHeaderField("Content-Length")).thenReturn(String.valueOf(contentLength));
+
+        //Mock Response to use sample json
+        BoxHttpResponse response = new BoxHttpResponse(huc);
+        PowerMockito.when(response.getContentLength()).thenReturn(contentLength);
```

Comment:
Looks like you are creating a new instance of `BoxHttpResponse` but you are still able to stub the result of methods. Is `new BoxHttpResponse` on line 62 returning a mock?

Reply:
PowerMockito.whenNew(BoxHttpResponse.class).withAnyArguments().thenReturn(response);
This will ensure that this response is returned.